builder/openstack/step_create_volume.go

```diff
@@ -82,6 +82,11 @@ func (s *StepCreateVolume) Run(ctx context.Context, state multistep.StateBag) mu
 		err := fmt.Errorf("Error waiting for volume: %s", err)
 		state.Put("error", err)
 		ui.Error(err.Error())
+
+		// The volume ID has created, try to clean up
+		s.doCleanup = true
```

Comment:
Just looking at it, it feels like `doCleanup` is a bit superfluous isn't it? I guess the `Cleanup` function could take a peek at `s.volumeID` and attempt to clean it up if it's not empty.